controller/engine_controller.go

```diff
@@ -266,26 +266,44 @@ func (ec *EngineController) syncEngine(key string) (err error) {
 		}
 	}()
 
+	var isCLIAPIVersionOne bool
+	if engine.Status.CurrentImage != "" {
```

Comment:
Why do we need to check the `CurrentImage`? The CurrentImage should be set if the volume is running.

Reply:
I want to make sure that the empty string won't be used in the function. Then I can guarantee that the related fields are cleaned up for the incompatible detached volume.

Reply:
I don't follow. Is `CurrentImage==""` valid or not? In which situation we will fall back to use `Spec.EngineImage`?

Reply:
Sometimes `CurrentImage` is cleaned up but `Endpoint` and `ReplicaModeMap` retain there. I will double-check the reason.

Reply:
Cannot reproduce it... It should be fine to just use `CurrentImage` only without `Spec.EngineImage` as fallback. But I am afraid that there may be unexpected corner cases that will cause the above bug.